Feature/footer

XDRO opened the pull request:
## Describe your changes
updated main to reflect remote branch. updated footer to add functionality inside of storybook, by adding controls to change the screenWidth and check mobile resolutions, rather than having to actually change the resolution. 
## Issue ticket number and link
EES-10
https://tripleten-apiary.atlassian.net/browse/EES-10
## Checklist before requesting a review

- [ ] I have performed a self-review of my code
- [ ] If there are changes to components, I have added / updated automated tests
- [ ] I have made any necessary updates to Storybook to accompany these changes
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings

Commit:
create new branch after merging local main to remote main for footer

Commit:
update Footer with conditional logic to give storybook users more control over the compoenent

RPinkha commented on tailwind.config.js:
I think it would be better to pull these changes from main now that they are merged.

XDRO replied:
I agree, but I'm not sure that I can do that since I made the branch before these changes were merged into main. I might have to look into a way to do this, without creating a new branch but honestly I don't believe that it'd cause any merge conflict if they are the exact same. If it does become an issue I will just create a new branch again.

RPinkha reviewed (commented):
Good job, you are almost there!
There seems to be an issue still with the background color not appearing fully on the footer from 640px - 725px. 
![image](https://github.com/user-attachments/assets/3d822558-3461-44c4-8d90-c68cf851071c)

RPinkha commented:
> Good job, you are almost there! There seems to be an issue still with the background color not appearing fully on the footer from 640px - 725px. ![image](https://private-user-images.githubusercontent.com/15398254/366994741-3d822558-3461-44c4-8d90-c68cf851071c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjYxNjI0ODEsIm5iZiI6MTcyNjE2MjE4MSwicGF0aCI6Ii8xNTM5ODI1NC8zNjY5OTQ3NDEtM2Q4MjI1NTgtMzQ2MS00NGM0LThkOTAtYzY4Y2Y4NTEwNzFjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MTIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTEyVDE3Mjk0MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWIxMDQ4MWUxODUxMmEyMzNjMDYyNWVhYjExZjhhNTRlYWEzZTRiZjZjMDNlZGZmMWQ4YjY5NzQ5YWIxMTJlZDYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.fmW2DtD_rxxeyT0C-EUFyFMsibbVmqmKvYWdi151dw8)

One more thing here. From ~868px, "Rental Conditions" becomes 2 lines, and at around 816px "Social Media" becomes 2 lines. Maybe reduce the gap between the 2 columns to accommodate the second words not going on a second line.

Commit:
add styles to address pier review comments

XDRO commented:
> > Good job, you are almost there! There seems to be an issue still with the background color not appearing fully on the footer from 640px - 725px. ![image](https://private-user-images.githubusercontent.com/15398254/366994741-3d822558-3461-44c4-8d90-c68cf851071c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjYxNjI0ODEsIm5iZiI6MTcyNjE2MjE4MSwicGF0aCI6Ii8xNTM5ODI1NC8zNjY5OTQ3NDEtM2Q4MjI1NTgtMzQ2MS00NGM0LThkOTAtYzY4Y2Y4NTEwNzFjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MTIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTEyVDE3Mjk0MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWIxMDQ4MWUxODUxMmEyMzNjMDYyNWVhYjExZjhhNTRlYWEzZTRiZjZjMDNlZGZmMWQ4YjY5NzQ5YWIxMTJlZDYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.fmW2DtD_rxxeyT0C-EUFyFMsibbVmqmKvYWdi151dw8)
> 
> One more thing here. From ~868px, "Rental Conditions" becomes 2 lines, and at around 816px "Social Media" becomes 2 lines. Maybe reduce the gap between the 2 columns to accommodate the second words not going on a second line.
I have addressed these concerns

raymondafan commented on src/components/Footer/Footer.tsx:
```diff
@@ -0,0 +1,116 @@
+            >
+              Social Media
+            </p>
+            <div className="flex flex-row gap-[10px] cursor-pointer">
```
hey @XDRO I did notice that when hovering over the gaps between the social media images, the cursor: pointer is visible. I would recommend taking out the the gap and give each image a div tag with a margin right of 10px and a cursor pointer as well. Here is an example of what i did: 
<img width="1440" alt="Screenshot 2024-09-13 at 4 47 35 PM" src="https://github.com/user-attachments/assets/e1539f6f-0bed-476b-8faf-8638ab4d768d">

raymondafan reviewed (commented):
Code looks great I really liked the way you coded the mobile version of the site. I did have one suggestion about the social media images and the cursor but other than that, this should be good to go.

RPinkha committed:
fix: fix some footer responsiveness

RPinkha committed:
fix: remove comments

RPinkha committed:
fix: add isMobile for footer element

RPinkha committed:
fix: push to resolve conflicts

Commit:
add cursor pointer to img element per pier review, remove gap from isMobile

Commit:
prioritize gap 26px in conditional argument isMobile with exclamation point

RPinkha committed:
fix: removed comment

RPinkha reviewed (approved):
Looks good on my end. 
Approved!

raymondafan reviewed (approved):
everything looks good on storybook the big gap is removed and the cursor pointer is only focused on the social media icons. this is ready for final review

RPinkha committed:
Merge branch 'main' into feature/footer-new

RPinkha committed:
fix: removed changes to main.tsx

tomershvueli commented on src/components/Footer/Footer.tsx:
```diff
@@ -0,0 +1,122 @@
+              className={`font-text text-paragraph-sm font-normal leading-[26px] tracking-[-0.02em] text-left sm:text-center cursor-pointer ${
+                isMobile ? "text-center" : "text-left"
+              }`}
+              onClick={() => console.log("clicked address")}
```
Let's get rid of all the `console.log`'s we have here, we don't want these showing up in production.

tomershvueli commented on src/components/Footer/Footer.tsx:
```diff
@@ -0,0 +1,122 @@
+              }`}
+              onClick={() => console.log("clicked email")}
+            >
+              Info@experteventsolutions.com
```
Can we make this a `mailto:` link?

tomershvueli commented on src/components/Footer/Footer.tsx:
```diff
@@ -0,0 +1,122 @@
+          isMobile ? "text-center" : "text-left"
+        }`}
+      >
+        Copyright © 2022. Expert Event Solutions - All rights reserved.
```
Neat trick here is to plug in Javascript for the year, ie `(new Date).getFullYear()` so this doesn't need updating every year.

RPinkha committed:
fix: removed all console logs

RPinkha committed:
feat: added mailto to the email field and changed to a tag

RPinkha committed:
feat: changed the year to be updating

RPinkha commented:
@tomershvueli all comments were addressed.

RPinkha committed:
Merge branch 'main' into feature/footer-new

RPinkha committed:
feat: made header fixed and removed comments

tomershvueli reviewed (approved):
LGTM!
